### ISSUE TITLE
Nicehash / Hashrate rental compatibility

### ISSUE THREAD

**Issue by dh35** [ASICseer/asicseer-pool#5]
**Environment**
Devices: Nicehash rental dashboard

**Steps to Reproduce**
1. Have ASICseer ops connect Nicehash accounts to pool and verify compatibility
2. Mine working shares with Nicehash on [ASICseer BCH pool]

**Expected Result**
Miners are able to rent hashrate from Nicehash (or similar rental platforms) with full compatibility and without a massive negative delta (rejects). 

Pool software should give nicehash the correct starting stratum difficulty (>500,000) to allow connection, and nicehash difficulty must always remain >500,000 or Nicehash will cancel order. Vardiff is okay, but vardiff floor for nicehash must be 500,001.

**Comment by tfaoro** [comment]
A potential solution here would be rework the "trusted" and "serverurl" config options to each have their own independent mindiff, startdiff, and maxdiff. You can already configure multiple ports on a single instance, so doing this would allow each port to have different diff parameters. For example:

Current config:
```
{
"trusted" : [
      "0.0.0.0:3334"
],
"mindiff" : 500001,
"startdiff" : 750000,
"maxdiff" : 0,
}
```
 Proposed config:
```
"trusted" : [
{
   "name": "standard",
   "mindiff": "50",
   "startdiff": "50000",
   "maxdiff": "0",
   "bind": "0.0.0.0:3333"
},
{
   "name": "nicehash",
   "mindiff": "500001",
   "startdiff": "750000",
   "maxdiff": "0",
   "bind": "0.0.0.0:3334"
}
]
```

**Comment by cculianu** [comment]
We went with the `mindiff_overrides` approach as discussed in Telegram. See #13 .

This should be fixed now by #13.  We will wait for more tests and then close this issue after 100% confirmation that it's running as intended.

**Comment by dh35** [comment]
Working as intended.
